src/utils/content-size.test.ts

```diff
@@ -0,0 +1,85 @@
+import { expect } from "chai"
+import { estimateContentSize, estimateFileSize, estimateTokens, wouldExceedSizeLimit } from "./content-size"
+import fs from "fs/promises"
+import path from "path"
+import os from "os"
+
+const CONTEXT_LIMIT = 1000 // Context limit of 1000 tokens means max allowed size is 500 tokens
+
+describe("content-size", () => {
+	describe("estimateTokens", () => {
+		it("estimates tokens based on byte count", () => {
+			expect(estimateTokens(100)).to.equal(25) // 100 bytes / 4 chars per token = 25 tokens
+			expect(estimateTokens(7)).to.equal(2) // Should round up for partial tokens
+		})
+	})
+
+	describe("wouldExceedSizeLimit", () => {
+		it("checks if byte count would exceed max allowed size", () => {
+			// For deepseek (64k - 27k = 37k tokens)
+			expect(wouldExceedSizeLimit(100, 37_000)).to.equal(false) // 25 tokens < 37k tokens
+			expect(wouldExceedSizeLimit(148000, 37_000)).to.equal(true) // 37k tokens = 37k tokens
+			// For standard models (128k - 30k = 98k tokens)
+			expect(wouldExceedSizeLimit(392000, 98_000)).to.equal(true) // 98k tokens = 98k tokens
+		})
+	})
+
+	describe("estimateContentSize", () => {
+		it("estimates size for string content", () => {
+			const content = "Hello world" // 11 bytes
+			const result = estimateContentSize(content, CONTEXT_LIMIT)
+
+			expect(result.bytes).to.equal(11)
+			expect(result.estimatedTokens).to.equal(3)
+			expect(result.wouldExceedLimit).to.equal(false)
+		})
+
+		it("estimates size for buffer content", () => {
+			const content = Buffer.from("Hello world") // 11 bytes
+			const result = estimateContentSize(content, CONTEXT_LIMIT)
+
+			expect(result.bytes).to.equal(11)
+			expect(result.estimatedTokens).to.equal(3)
+			expect(result.wouldExceedLimit).to.equal(false)
+		})
+
+		it("detects when content would exceed max allowed size", () => {
+			// Create content that would exceed max allowed size for deepseek (64k - 27k tokens)
+			const largeContent = "x".repeat(148000) // 37k tokens > (64k - 27k) tokens
+			const result = estimateContentSize(largeContent, 64_000)
+
+			expect(result.wouldExceedLimit).to.equal(true)
+		})
+	})
+
+	describe("estimateFileSize", () => {
+		let tempFilePath: string
+
+		beforeEach(async () => {
+			tempFilePath = path.join(os.tmpdir(), "test-file.txt")
+			await fs.writeFile(tempFilePath, "Hello world") // 11 bytes
+		})
+
+		afterEach(async () => {
+			await fs.unlink(tempFilePath).catch(() => {})
+		})
+
+		it("estimates size for existing file", async () => {
+			const result = await estimateFileSize(tempFilePath, CONTEXT_LIMIT)
+
+			expect(result.bytes).to.equal(11)
+			expect(result.estimatedTokens).to.equal(3)
+			expect(result.wouldExceedLimit).to.equal(false)
+		})
+
+		it("throws error for non-existent file", async () => {
```

Comment:
I don't understand this test. Is the `await estimateFileSize(nonExistentPath, CONTEXT_LIMIT)` expected to throw? 
This test will always pass right? because if `await estimateFileSize(nonExistentPath, CONTEXT_LIMIT)` succeeds we still throw an error `throw new Error("Should have thrown error")` making the test useless.

Reply:
In `estimateFileSize()`, note that it uses `fs.stat()` to get file information. When a file doesn't exist, `fs.stat()` will throw an error. This means `estimateFileSize()` should throw an error naturally when given a non-existent file path.

The test is structured using a try-catch block with the following logic:

- Try to call estimateFileSize() on a non-existent file
- If estimateFileSize() throws (expected behavior), the catch block will run and verify the error
- If estimateFileSize() somehow succeeds (unexpected behavior), it will throw a new error with "Should have thrown error"